predicators/datasets/vlm_input_data.py

```diff
@@ -1,17 +1,245 @@
+    relevant for planning to recreate these trajectories."""
+    aggregated_vlm_output_strs = []
+    all_vlm_queries_list = []
+    for traj in trajectories:
+        all_vlm_queries_list += _generate_prompt_for_atom_proposals(
+            traj, trajectory_subsample_freq)
+    curr_num_queries = 0
+    total_num_queries = len(all_vlm_queries_list)
+    for query in all_vlm_queries_list:
+        aggregated_vlm_output_strs.append(
+            vlm.sample_completions(query[0],
+                                   query[1],
+                                   0.0,
+                                   CFG.seed,
+                                   num_completions=1))
+        curr_num_queries += 1
+        logging.info("Completed (%s/%s) init atoms queries to the VLM.",
+                     curr_num_queries, total_num_queries)
+    return aggregated_vlm_output_strs
+
+
+def _label_trajectories_with_atom_values(
+        trajectories: List[ImageOptionTrajectory], vlm: VisionLanguageModel,
+        atoms_list: List[str]) -> List[List[str]]:
+    """Given a list of atoms, label every state in ImageOptionTrajectories with
+    the truth values of a set of atoms."""
+    total_scenes_to_label = sum(len(traj.imgs) for traj in trajectories)
+    curr_scenes_labelled = 0
+    output_labelled_atoms_txt_list = []
+    for traj in trajectories:
+        prompts_for_traj = _generate_prompt_for_scene_labelling(
+            traj, atoms_list)
+        curr_traj_txt_outputs = []
+        for prompt in prompts_for_traj:
+            # Sample VLM outputs with temperature 0 in an attempt to be
+            # accurate.
+            curr_vlm_atom_labelling = vlm.sample_completions(prompt[0],
+                                                             prompt[1],
+                                                             0.0,
+                                                             CFG.seed,
+                                                             num_completions=1)
+            assert len(curr_vlm_atom_labelling) == 1
+            sanitized_output = curr_vlm_atom_labelling[0].replace('\\', '')
+            curr_traj_txt_outputs.append(sanitized_output)
+            curr_scenes_labelled += 1
+            logging.info("Completed (%s/%s) label queries to VLM!",
+                         curr_scenes_labelled, total_scenes_to_label)
+        output_labelled_atoms_txt_list.append(curr_traj_txt_outputs)
+    return output_labelled_atoms_txt_list
+
+
+def _parse_unique_atom_proposals_from_list(
+        atom_strs_proposals_list: List[List[str]],
+        relevant_objects_across_demos: Set[Object]) -> Set[str]:
+    """Given a list of atom proposals that a VLM has constructed for each
+    demonstration, parse these to a unique set of proposals.
+
+    This function currently does 3 steps of sanitization: (1) removing
+    any unnecessary characters, (2) removing any atoms that involve
+    objects that aren't relevant, (3) removing any duplicate atoms.
```

Comment:
can you clarify what "relevant" means? is it more like "known"?

Reply:
also why are we considering uniqueness at the level of atoms instead of predicates?